File: 2640-maximum-number-of-integers-to-choose-from-a-range-i/maximum-number-of-integers-to-choose-from-a-range-i.cpp
class Solution {
public:
    int maxCount(vector<int>& banned, int n, int maxSum) {
        
        set<int>arr;
        for (int i=0;i<banned.size();i++)
        {
            arr.insert(banned[i]);
        }
        int sum=0,count=0;
         for (int i=1;i<=n;i++)
            {
                if(arr.find(i)== arr.end() && sum+i <= maxSum)
                {
                    sum+=i;
                      count++;
                    
                }
            }
        return count;
    }
};

File: 2640-maximum-number-of-integers-to-choose-from-a-range-i/maximum-number-of-integers-to-choose-from-a-range-i.java
import java.util.*;
class Solution {
    public int maxCount(int[] banned, int n, int maxSum) {
        
        Set<Integer> ar = new HashSet<>();

        for (int i : banned)
        {
            ar.add(i);
        }
        int j=0,sum=0,count=0;
        for (int i=1;i<=n;i++)
        {
            if(!ar.contains(i)&& sum+i <= maxSum)
            {
                sum+=i;
                count++;
            }
        }
       
       return count;
    }
}